bumble/core.py

```diff
@@ -288,8 +288,19 @@ def __str__(self):
 
         return result
 
-    def __repr__(self):
-        return str(self)
+    def __repr__(self) -> str:
+        if len(self.uuid_bytes) == 2 or len(self.uuid_bytes) == 4:
+            return bytes(reversed(self.uuid_bytes)).hex().upper()
+
+        return '-'.join(
```

Comment:
Since this is the same as what's used in `__str__`, consider factoring out this full-size uuid->string conversion in its own method.
Maybe all we need is to add a `separator` argument to `to_hex_string` with a default of '', then `__repr__` becomes just `self.to_hex_string(separator="-")`, and `__str__` would also call `to_hex_string(separator="-")` for 16-byte UUIDs.